Muhammadk 03/fix llamadev no tests skip

muhammadkhalid-03 opened the pull request:
# Description

This PR is for the wrong error that was being printed in stderr. Specifically, when a package had no tests, llamadev still reported the error "1 packages were skipped due to Python version incompatibility:".
This fix creates two different arrays. One of them keeps track of the packages skipped due to version incompatibility and the other keeps track of packages skipped due to no tests being present and prints the appropriate message to stderr.

Fixes #19420 

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [x] I added new unit tests to cover this change
- [ ] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `uv run make format; uv run make lint` to appease the lint gods

masci commented on llama-dev/llama_dev/test/__init__.py:
```diff
@@ -157,17 +157,29 @@ def test(
+    skipped_no_tests = [
+        r["package"].relative_to(repo_root)
+        for r in results
+        if r["status"] == ResultStatus.SKIPPED and "package has no tests" in r["stderr"]
```
While this works, the root problem is that `ResultStatus.SKIPPED` is returned both when `if not package_has_tests(package_path):` and when `if not is_python_version_compatible(package_data):` at lines 284 and 294. 

If we replace `ResultStatus.SKIPPED` with two new statuses like `NO_TESTS` and `UNSUPPORTED_PYTHON_VER` (or something like this), this check would be more robust.

masci reviewed (changes requested):
The logic looks good to me, it would be cool to make the check more robust by using dedicated result statuses instead of parsing the error message

muhammadkhalid-03 commented:
I've added the two new statuses and changed the CLI output message. I also updated the tests for the two new statuses. They use the CLI output message as well as the status for testing. Let me know if there's anything I still need to change!

masci reviewed (approved):
LGTM, thanks for the fix!
